ambari-server/src/main/java/org/apache/ambari/server/api/handlers/DeleteHandler.java

```diff
@@ -83,8 +83,8 @@ protected ResultMetadata convert(RequestStatusMetaData requestStatusMetaData) {
       return null;
     }
 
-    if (requestStatusMetaData.getClass() != DeleteStatusMetaData.class) {
-      throw new IllegalArgumentException("RequestStatusDetails is not of type DeleteStatusDetails");
+    if (!DeleteStatusMetaData.class.isInstance(requestStatusMetaData)) {
+      throw new IllegalArgumentException("RequestStatusDetails is not of type DeleteStatusDetails. requestStatusMetaData.getClass= " + requestStatusMetaData.getClass());
```

Comment:
Exception message is not very much descriptive, let's make it more clear: 
`String.format("Wrong status details received -  expecting: %s; actual: %s", <class name>, <class name>)`

Reply:
Done. Thanks